pkg/infrastructure/services/prometheus.go

```diff
@@ -18,66 +18,157 @@ import (
-func CreateServiceMonitor(kogitoRuntime *v1alpha1.KogitoRuntime) *monv1.ServiceMonitor {
-	monitoring := kogitoRuntime.Spec.Monitoring
-	if monitoring.Scrape {
-		endPoint := monv1.Endpoint{}
-		endPoint.TargetPort = &intstr.IntOrString{IntVal: getServiceHTTPPort(kogitoRuntime)}
-
-		if len(monitoring.Path) > 0 {
-			endPoint.Path = monitoring.Path
-		} else {
-			endPoint.Path = v1alpha1.MonitoringDefaultPath
-		}
+func isPrometheusAddOnAvailable(client *client.Client, kogitoService v1alpha1.KogitoService) (bool, error) {
+	url, err := infrastructure.FetchKogitoServiceURI(client, kogitoService.GetName(), kogitoService.GetNamespace())
+	if err != nil {
+		return false, err
+	}
+
+	url = url + getMonitoringPath(kogitoService.GetSpec().GetMonitoring())
+	if resp, err := http.Get(url); err != nil {
+		return false, err
+	} else if resp.StatusCode == http.StatusOK {
+		return true, nil
+	}
+	log.Debugf("Non-OK Http Status received")
+	return false, nil
+}
 
-		if len(monitoring.Scheme) > 0 {
-			endPoint.Scheme = monitoring.Scheme
-		} else {
-			endPoint.Scheme = v1alpha1.MonitoringDefaultScheme
+func createPrometheusServiceMonitorIfNotExists(client *client.Client, kogitoService v1alpha1.KogitoService, scheme *runtime.Scheme) error {
+	serviceMonitor, err := loadDeployedServiceMonitor(client, kogitoService.GetName(), kogitoService.GetNamespace())
+	if err != nil {
+		return err
+	}
+	if serviceMonitor == nil {
+		_, err := createServiceMonitor(client, kogitoService, scheme)
+		if err != nil {
+			return err
 		}
+	}
+	return nil
+}
 
-		serviceSelectorLabels := make(map[string]string)
-		serviceSelectorLabels[framework.LabelAppKey] = kogitoRuntime.GetName()
+func loadDeployedServiceMonitor(client *client.Client, instanceName, namespace string) (*monv1.ServiceMonitor, error) {
+	log.Debug("fetching deployed Service monitor instance with name %s in namespace %s", instanceName, namespace)
+	serviceMonitor := &monv1.ServiceMonitor{}
+	if exits, err := kubernetes.ResourceC(client).FetchWithKey(types.NamespacedName{Name: instanceName, Namespace: namespace}, serviceMonitor); err != nil {
+		log.Error("Error occurs while fetching Service monitor instance")
+		return nil, err
+	} else if !exits {
+		log.Debug("Service monitor instance is not exists")
+		return nil, nil
+	} else {
+		log.Debug("Service monitor instance found")
+		return serviceMonitor, nil
+	}
+}
 
-		serviceMonitorLabels := make(map[string]string)
-		serviceMonitorLabels["name"] = operator.Name
-		serviceMonitorLabels[framework.LabelAppKey] = kogitoRuntime.GetName()
+// createServiceMonitor create ServiceMonitor used for scraping by prometheus for kogito service
+func createServiceMonitor(cli *client.Client, kogitoService v1alpha1.KogitoService, scheme *runtime.Scheme) (*monv1.ServiceMonitor, error) {
+	monitoring := kogitoService.GetSpec().GetMonitoring()
+	endPoint := monv1.Endpoint{}
+	endPoint.TargetPort = &intstr.IntOrString{IntVal: getServiceHTTPPort(kogitoService)}
```

Comment:
I haven't found the `getServiceHTTPPort` function in this PR, but just make sure we are returning the `service.Port` and not `service.TargetPort`.

Reply:
IMHO the `getServiceHTTPPort` returns a port where Kogito application is running (a bit unfortunate naming) - https://github.com/kiegroup/kogito-cloud-operator/blob/master/pkg/infrastructure/services/deployment.go#L85. 
Needs to be changed here to use service port.

Reply:
Yes, this function is getting the port from the spec, getting from the service directly will be more reliable.

Reply:
@ricardozanini @sutaakar 
Previously I done so because exposed port are always equal to kogitoService port but now after implementation of KOGITO-3516, exposed port always be a default port. So rather then loading Service object and fetch port from it, could I directly use default http?
We could use same approach to create kogitoServiceURI as well.
https://github.com/kiegroup/kogito-cloud-operator/blob/605cb71f5d01b94de56516aaa230ec37232fb846/pkg/infrastructure/kogito_service.go#L110

Reply:
+1 @vaibhavjainwiz

Reply:
done